provider.go

```diff
@@ -137,47 +205,37 @@ func (p *provider) DeleteServicePrincipal(ctx context.Context, spObjectID string
 }
 
 // ListRoles like all Azure roles with a scope (often subscription).
-func (p *provider) ListRoleDefinitions(ctx context.Context, scope string, filter string) (result []authorization.RoleDefinition, err error) {
-	page, err := p.rdClient.List(ctx, scope, filter)
-
+func (p *provider) ListRoleDefinitions(ctx context.Context, scope string, filter string) (result []*armauthorization.RoleDefinition, err error) {
+	options := armauthorization.RoleDefinitionsClientListOptions{
+		Filter: &filter,
+	}
+	page := p.rdClient.NewListPager(scope, &options)
+	listResp, err := page.NextPage(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	return page.Values(), nil
+	return listResp.Value, err
 }
 
 // GetRoleByID fetches the full role definition given a roleID.
-func (p *provider) GetRoleDefinitionByID(ctx context.Context, roleID string) (result authorization.RoleDefinition, err error) {
-	return p.rdClient.GetByID(ctx, roleID)
+func (p *provider) GetRoleDefinitionByID(ctx context.Context, roleID string) (result armauthorization.RoleDefinitionsClientGetByIDResponse, err error) {
+	return p.rdClient.GetByID(ctx, roleID, nil)
 }
 
 // CreateRoleAssignment assigns a role to a service principal.
-func (p *provider) CreateRoleAssignment(ctx context.Context, scope string, roleAssignmentName string, parameters authorization.RoleAssignmentCreateParameters) (authorization.RoleAssignment, error) {
-	return p.raClient.Create(ctx, scope, roleAssignmentName, parameters)
+func (p *provider) CreateRoleAssignment(ctx context.Context, scope string, roleAssignmentName string, parameters armauthorization.RoleAssignmentCreateParameters) (armauthorization.RoleAssignmentsClientCreateResponse, error) {
+	return p.raClient.Create(ctx, scope, roleAssignmentName, parameters, nil)
 }
 
 // GetRoleAssignmentByID fetches the full role assignment info given a roleAssignmentID.
-func (p *provider) GetRoleAssignmentByID(ctx context.Context, roleAssignmentID string) (result authorization.RoleAssignment, err error) {
-	return p.raClient.GetByID(ctx, roleAssignmentID)
+func (p *provider) GetRoleAssignmentByID(ctx context.Context, roleAssignmentID string) (armauthorization.RoleAssignmentsClientGetByIDResponse, error) {
+	return p.raClient.GetByID(ctx, roleAssignmentID, nil)
 }
 
 // DeleteRoleAssignmentByID deletes a role assignment.
-func (p *provider) DeleteRoleAssignmentByID(ctx context.Context, roleAssignmentID string) (result authorization.RoleAssignment, err error) {
-	return p.raClient.DeleteByID(ctx, roleAssignmentID)
-}
-
-// ListRoleAssignments lists all role assignments.
-// There is no need for paging; the caller only cares about the first match and whether
-// there are 0, 1 or >1 items. Unpacking here is a simpler interface.
-func (p *provider) ListRoleAssignments(ctx context.Context, filter string) ([]authorization.RoleAssignment, error) {
```

Comment:
Did not see thing being used anywhere, opted to remove

Reply:
Tracked this back a bit and looks like it was used for testing before. Good with removing it 👍